File: backend/src/main/java/com/agentos/model/entity/Agent.java
package com.agentos.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("agent")
public class Agent {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String description;

    private String systemPrompt;

    private String modelProvider;

    private String modelName;

    private Double temperature;

    private Integer maxTokens;

    private String avatarUrl;

    private Long ownerId;

    private Integer status;

    // FieldStrategy.NEVER 表示该字段不允许插入和更新
    @TableField(insertStrategy = FieldStrategy.NEVER, updateStrategy = FieldStrategy.NEVER)
    private LocalDateTime createdAt;

    @TableField(insertStrategy = FieldStrategy.NEVER, updateStrategy = FieldStrategy.NEVER)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/agent.ts
// Agent 实体（对应后端 Agent 实体）
export interface Agent {
  id: number
  name: string
  description: string | null
  systemPrompt: string
  modelProvider: string | null
  modelName: string
  temperature: number | null
  maxTokens: number | null
  avatarUrl: string | null
  ownerId: number
  status: number | null  // 0=stopped, 1=running
  createdAt: string
  updatedAt: string
}

// Agent 新建/编辑表单数据（对应 CreateAgentReq / UpdateAgentReq）
export interface AgentFormData {
  name: string
  description?: string
  systemPrompt: string
  modelProvider?: string
  modelName: string
  temperature?: number
  maxTokens?: number
}
